File: go/71_SimplifyPath.go
package main

import (
    "strings"
)

func simplifyPath(path string) string {
    stacks := []string{}
    for _, v := range strings.Split(path, "/") {
        if v == "." || v == "" {
            continue
        } else if v == ".." {
            if len(stacks) > 0 {
                stacks = stacks[:len(stacks)-1]
            }
        } else {
            stacks = append(stacks, v)
        }
    }
    return "/" + strings.Join(stacks, "/")
}

File: python/71_SimplifyPath.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stacks = []
        for v in path.split("/"):
            if not v or v == ".":
                continue
            elif v == "..":
                if stacks:
                    stacks.pop(len(stacks) - 1)
            else:
                stacks.append(v)
        return "/" + "/".join(stacks)
